front/app/components/IdeaCards/TopicFilterDropdown.tsx

```diff
@@ -1,81 +1,82 @@
 import React, { memo, useState, useMemo } from 'react';
-import { isNilOrError } from 'utils/helperUtils';
 
 // components
 import FilterSelector from 'components/FilterSelector';
 
 // services
-import { ITopicData } from 'services/topics';
+import { getTopicIds } from 'services/projectAllowedInputTopics';
 
 // i18n
 import injectLocalize, { InjectedLocalized } from 'utils/localize';
 import { FormattedMessage } from 'utils/cl-intl';
 import messages from './messages';
 
 // hooks
+import useProjectAllowedInputTopics from 'hooks/useProjectAllowedInputTopics';
 import useTopics from 'hooks/useTopics';
 
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+
 interface Props {
   alignment: 'left' | 'right';
   onChange: (value: any) => void;
-  projectId: string | null;
+  projectId: string;
 }
 
 const TopicFilterDropdown = memo(
   ({ alignment, projectId, onChange, localize }: Props & InjectedLocalized) => {
     const [selectedValues, setSelectedValues] = useState<string[]>([]);
-    const topics = useTopics(projectId ? { projectId } : {});
+    const allowedInputTopics = useProjectAllowedInputTopics(projectId);
+
+    const topicIds = useMemo(
```

Comment:
Wondering how useful is the useMemo as `const allowedInputTopics` is not in itself memoized as far as I can see.

Also, don't think it's a problem if this is not memoized, probably.